javalin-utils/javalin-context-mock/src/main/java/io/javalin/mock/ContextMockConfigurer.kt

```diff
@@ -0,0 +1,10 @@
+package io.javalin.mock
+
+fun interface ContextMockConfigurer {
+    /** Apply changes to the [ContextMockConfig] instance. */
+    fun ContextMockConfig.configure()
+}
+
+internal fun invokeMockConfigurerWithAsSamWithReceiver(fn: ContextMockConfigurer, receiver: ContextMockConfig) {
+    with(fn) { receiver.configure() }
+}
```

Comment:
Keep in mind that it's impossible to call this function without this utility function:

```kotlin
fun interface ContextMockConfigurer {
    fun ContextMockConfig.configure() // extension function on ContextMockConfig in ContextMockConfigurer scope
}
```

Because:

```kotlin
fn.configure() // does not exist, we can't pass ContextMockConfig
```

The `with(ContextMockConfigurer)` call changes current scope, so it's executed within the `fn` (== `this`). In Java it looks like a regular function tho, so it's not an issue.

Reply:
I would probably name it something like:

```kotlin
internal fun invokeConfigWithConfigurerScope(configurer: ContextMockConfigurer, config: ContextMockConfig) {
    with(configurer) { config.configure() }
}
```

I've already approved the changes though, so just let me know when you want to merge 😊

Reply:
Cool, I don't mind - it's only for our internal usage anyway :)